code/modules/library/library_catalog.dm

```diff
@@ -0,0 +1,617 @@
+	report_types = list(
+		DEFINE_CATEGORY(LIB_REPORT_HATESPEECH, "Hatespeech or Slur Usage"),
+		DEFINE_CATEGORY(LIB_REPORT_EROTICA, "Erotica or Sexual Content"),
+		DEFINE_CATEGORY(LIB_REPORT_OOC, "Out of Character Information"),
+		DEFINE_CATEGORY(LIB_REPORT_COPYPASTA, "Copypastas or Spam"),
+		DEFINE_CATEGORY(LIB_REPORT_BLANK , "Blank or No Content"),
+		DEFINE_CATEGORY(LIB_REPORT_NOEFFORT , "Very Low Effort Content"),
+		DEFINE_CATEGORY(LIB_REPORT_OTHER  , "Other Reason not Specified"), //required
+	)
+
+	//building a list of all categories, used for searching, Cat ID needs to be unique to category
+	categories = list(
+		DEFINE_CATEGORY(LIB_CATEGORY_FICTION, "Fiction"),
+		DEFINE_CATEGORY(LIB_CATEGORY_NONFICTION, "Non-Fiction"),
+		DEFINE_CATEGORY(LIB_CATEGORY_RELIGION, "Religious"),
+		DEFINE_CATEGORY(LIB_CATEGORY_FANTASY, "Fantasy"),
+		DEFINE_CATEGORY(LIB_CATEGORY_HORROR, "Horror"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ROMANCE, "Romance"),
+		DEFINE_CATEGORY(LIB_CATEGORY_MYSTERY, "Mystery"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ADVENTURE, "Adventure"),
+		DEFINE_CATEGORY(LIB_CATEGORY_HISTORY, "History"),
+
+		DEFINE_CATEGORY(LIB_CATEGORY_PHILOSOPHY, "Philosophy"),
+		DEFINE_CATEGORY(LIB_CATEGORY_DRAMA, "Drama and Thriller"),
+    	DEFINE_CATEGORY(LIB_CATEGORY_EXPERIMENT, "Experiment Notes"),
+		DEFINE_CATEGORY(LIB_CATEGORY_LEGAL, "Legal Document"),
+		DEFINE_CATEGORY(LIB_CATEGORY_BIOGRAPHY, "Biography"),
+		DEFINE_CATEGORY(LIB_CATEGORY_GUIDE, "Guides and References"),
+		DEFINE_CATEGORY(LIB_CATEGORY_PAPERWORK, "Paperwork"),
+		DEFINE_CATEGORY(LIB_CATEGORY_COOKING, "Culinary Arts"),
+		DEFINE_CATEGORY(LIB_CATEGORY_DESIGN, "Decor and Design"),
+		DEFINE_CATEGORY(LIB_CATEGORY_COMBAT, "Martial Arts and Combat"),
+		DEFINE_CATEGORY(LIB_CATEGORY_EXPLORATION, "Exploration"),
+		DEFINE_CATEGORY(LIB_CATEGORY_THEATRE, "Theatre"),
+		DEFINE_CATEGORY(LIB_CATEGORY_POETRY, "Poetry"),
+
+    	DEFINE_CATEGORY(LIB_CATEGORY_LAW, "Law"),
+		DEFINE_CATEGORY(LIB_CATEGORY_SECURITY, "Security"),
+		DEFINE_CATEGORY(LIB_CATEGORY_SUPPLY, "Supply"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ENGINEERING, "Engineering"),
+		DEFINE_CATEGORY(LIB_CATEGORY_SERVICE , "Service"),
+		DEFINE_CATEGORY(LIB_CATEGORY_MEDICAL, "Medical"),
+		DEFINE_CATEGORY(LIB_CATEGORY_RESEARCH, "Science"),
+		DEFINE_CATEGORY(LIB_CATEGORY_COMMAND , "Command"),
+	)
+
+	//Books that we don't want showing up in the programmatic book list
+	//Books should go here if they're non-functional, spawners, or are designed for off-station roles to consume
+	var/list/forbidden_books = list(
+		/obj/item/book/manual/random,
+		/obj/item/book/manual/nuclear,
+		/obj/item/book/manual/wiki,
+		/obj/item/book/manual/hydroponics_pod_people,
+	)
+
+	var/newid = 1
+	//building a list of all programmatic books
+	for(var/typepath in (subtypesof(/obj/item/book/manual) - forbidden_books))
+		var/obj/item/book/B = new typepath(null)
+		var/datum/programmaticbook/PB = new()
+		PB.title = B.name
+		PB.author = B.author
```

Comment:
```suggestion
		var/obj/item/book/B = typepath
		var/datum/programmaticbook/PB = new()
		PB.title = initial(B.name)
		PB.author = initial(B.author)
```
This should work if those values are defined in the object definition. And not in the Initialize or such.